sql/parser/ast.go

```diff
@@ -205,22 +308,33 @@ const (
 	astDropTable       = "DROP TABLE"
 	astDropView        = "DROP VIEW"
 	astRenameTable     = "RENAME TABLE"
+	astShowDatabases   = "SHOW DATABASES"
 	astShowTables      = "SHOW TABLES"
 	astShowIndex       = "SHOW INDEX FROM"
 	astShowColumns     = "SHOW COLUMNS FROM"
 	astShowFullColumns = "SHOW FULL COLUMNS FROM"
 	astTruncateTable   = "TRUNCATE TABLE"
+	astIfNotExists     = " IF NOT EXISTS"
+	astUnique          = "UNIQUE"
+	astPrimaryKey      = "PRIMARY KEY"
+	astKey             = "KEY"
+	astUnsigned        = "UNSIGNED"
 )
 
 func (node *DDL) String() string {
 	switch node.Action {
-	case astCreateIndex, astDropIndex:
+	case astDropIndex:
 		return fmt.Sprintf("%s %s ON %s", node.Action, node.Name, node.NewName)
 	case astRenameTable:
 		return fmt.Sprintf("%s %s %s", node.Action, node.Name, node.NewName)
-	case astCreateDatabase, astCreateTable, astCreateView:
-		return fmt.Sprintf("%s %s", node.Action, node.NewName)
+	case astCreateView:
+		return fmt.Sprintf("%s%s %s", node.Action, node.IfNotExists, node.NewName)
+	case astShowDatabases:
```

Comment:
Hi,

Does cockroachDB have a concept of DATABASE?  if not, having a command like
SHOW DATABASES could be misleading.

The term DATABASE  is used very differently by various RDBMS vendors (e.g.
a DATABASE in context of Oracle is completely different to that in MS SQL
Server) hence ANSI SQL avoids this term and uses the following schema terms:


4.2.9 Catalogs and schemas
4.2.9.1 Catalogs
A catalog is a named collection of SQL-schemas, foreign server descriptors,
and foreign data wrapper
descriptors in an SQL-environment. The mechanisms for creating and
destroying catalogs are implementationdefined.


4.2.9.2 SQL-schemas
An SQL-schema, often referred to simply as a schema, is a persistent, named
collection of descriptors. Any
object whose descriptor is in some SQL-schema is known as an SQL-schema
object.
A schema, the schema objects in it, and the SQL-data described by them are
said to be owned by the authorization
identifier associated with the schema.
SQL-schemas are created and destroyed by execution of SQL-schema statements
(or by implementation-defined
mechanisms).


4.2.12 SQL-data
SQL-data is data described by SQL-schemas — data that is under the control
of an SQL-implementation in an
SQL-environment.

4.3 Tables
A table is a collection of zero or more rows where each row is a sequence
of one or more column values. In
general, rows in a table are unordered; however, rows in a table are
ordered if the table is the result of a <query
expression> that immediately contains an <order by clause>. Each column has
a name and a data type. Each
row has exactly one value for each column C, and the data type of that
value is the data type of C.


thank you,

Jusong Chen


On Mon, Jun 22, 2015 at 7:00 AM, Peter Mattis <notifications@github.com>
wrote:

> In sql/parser/ast.go
> <https://github.com/cockroachdb/cockroach/pull/1430#discussion_r32936338>:
>
> >  		return fmt.Sprintf("%s %s ON %s", node.Action, node.Name, node.NewName)
> >  	case astRenameTable:
> >  		return fmt.Sprintf("%s %s %s", node.Action, node.Name, node.NewName)
> > -	case astCreateDatabase, astCreateTable, astCreateView:
> > -		return fmt.Sprintf("%s %s", node.Action, node.NewName)
> > +	case astCreateView:
> > +		return fmt.Sprintf("%s%s %s", node.Action, node.IfNotExists, node.NewName)
> > +	case astShowDatabases:
>
> The vitess sql parser parses the MySQL variant of SQL. SHOW DATABASES
> comes from https://dev.mysql.com/doc/refman/5.1/en/show-databases.html.
> Namespace is not a term that is present in SQL. I'm not settled on this
> terminology, but it seems good to stick with the existing language for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1430/files#r32936338>.
>



-- 
Jusong Chen
408 315 2288